src/Frontend/Core/Engine/Model.php

```diff
@@ -326,6 +329,29 @@ public static function getModuleSettings($module)
     }
 
     /**
+     * Creates router for module based on routing file
+     *
+     * @param $module
+     * @return null|Router
+     */
+    public static function getModuleRouter($module)
```

Comment:
I will look into this, but i tried to think of something. First of all problem is that getURLForBlock method is used by spoon tampate and static module models. So there is not so many option. Maybe we could use singleton pattern and use getInstance when we need it, because i dont think that creating new module instance every time we need config is a smart solution.